src/Renci.SshNet/Security/Cryptography/Ciphers/AesCipher.cs

```diff
@@ -41,30 +33,22 @@ public enum AesCipherMode
     /// <summary>
     /// AES cipher implementation.
     /// </summary>
-    public sealed class AesCipher : BlockCipher, IDisposable
+    public sealed partial class AesCipher : BlockCipher, IDisposable
     {
-        private const int BLOCKSIZE = 16;
-
-        private readonly AesCipherMode _aesMode;
-        private readonly CipherMode _blockMode;
-
-        private readonly Aes _aes;
-        private ICryptoTransform _encryptor;
-        private ICryptoTransform _decryptor;
-
-        private AesCipher _ecbHelper;
-        private uint[] _packedIV;
+        private readonly BlockCipher _impl;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="AesCipher"/> class in ECB mode.
+        /// Initializes a new instance of the <see cref="AesCipher"/> class.
         /// </summary>
         /// <param name="key">The key.</param>
-        /// <param name="pkcs7Padding">Enable PKCS7 padding.</param>
+        /// <param name="mode">The mode.</param>
+        /// <param name="padding">The padding.</param>
         /// <exception cref="ArgumentNullException"><paramref name="key"/> is <see langword="null"/>.</exception>
         /// <exception cref="ArgumentException">Keysize is not valid for this algorithm.</exception>
-        public AesCipher(byte[] key, bool pkcs7Padding = false)
-            : this(key, iv: null, AesCipherMode.ECB, pkcs7Padding)
+        public AesCipher(byte[] key, CipherMode mode, CipherPadding padding)
+            : base(key, 16, mode, padding)
```

Comment:
This constructor should be removed, it's only currently used in the Tests project.

Reply:
Well it is public, so we don't know how many people are using it. But ok, let's see if the others are ok with it.